toolkit/tools/imagecustomizer/api/systemconfig.go

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerapi
+
+import (
+	"gopkg.in/yaml.v3"
+)
+
+type FileConfigList []FileConfig
+
+// SystemConfig defines how each system present on the image is supposed to be configured.
+type SystemConfig struct {
+	AdditionalFiles map[string]FileConfigList `yaml:"AdditionalFiles"`
+}
+
+// Configuration of files that are to be added to the image.
+type FileConfig struct {
+	DestinationFile string `yaml:"DestinationFile"`
+	FilePermissions string `yaml:"FilePermissions"`
+}
+
+// Allow FileConfigList to be parsed from just a single value.
+func (l *FileConfigList) UnmarshalYAML(value *yaml.Node) error {
```

Comment:
This is kind of cool in that I can maintain backwards-compatibility with the existing `AdditionalFiles` schema while adding support for both setting the file permissions and for a single source file being copied to 2 different locations.